test/integration/users.test.js

```diff
@@ -1071,4 +1076,156 @@ describe("Users", function () {
         });
     });
   });
+
+  describe("PATCH /users/:userId/roles", function () {
+    it("Should make the user a member", function (done) {
+      addUser(userRoleUpdate).then((userRoleUpdateId) => {
+        chai
+          .request(app)
+          .patch(`/users/${userRoleUpdateId}/roles`)
+          .set("cookie", `${cookieName}=${superUserAuthToken}`)
+          .send({
+            member: true,
+          })
+          .end((err, res) => {
+            if (err) {
+              return done(err);
+            }
+
+            expect(res).to.have.status(204);
+            /* eslint-disable no-unused-expressions */
+            expect(res.body).to.be.a("object").to.be.empty;
+            return done();
+          });
+      });
+    });
+
+    it("Should make the member a user", function (done) {
+      addUser(userRoleUpdate).then((userRoleUpdateId) => {
+        chai
+          .request(app)
+          .patch(`/users/${userRoleUpdateId}/roles`)
+          .set("cookie", `${cookieName}=${superUserAuthToken}`)
+          .send({
+            member: false,
+          })
+          .end((err, res) => {
+            if (err) {
+              return done(err);
+            }
+
+            expect(res).to.have.status(204);
+            /* eslint-disable no-unused-expressions */
+            expect(res.body).to.be.a("object").to.be.empty;
+            return done();
+          });
+      });
+    });
+
+    it("Should archive the user", function (done) {
+      addUser(userRoleUpdate).then((userRoleUpdateId) => {
+        chai
+          .request(app)
+          .patch(`/users/${userRoleUpdateId}/roles`)
+          .set("cookie", `${cookieName}=${superUserAuthToken}`)
+          .send({
+            archived: true,
+          })
+          .end((err, res) => {
+            if (err) {
+              return done(err);
+            }
+
+            expect(res).to.have.status(204);
+            /* eslint-disable no-unused-expressions */
+            expect(res.body).to.be.a("object").to.be.empty;
+            return done();
+          });
+      });
+    });
+
+    it("Should return 400 if user is already a member", function (done) {
+      addUser(userAlreadyMember).then((userAlreadyMemberId) => {
+        chai
+          .request(app)
+          .patch(`/users/${userAlreadyMemberId}/roles`)
+          .set("cookie", `${cookieName}=${superUserAuthToken}`)
+          .send({
+            member: true,
+          })
+          .end((err, res) => {
+            if (err) {
+              return done(err);
+            }
+
+            expect(res).to.have.status(400);
+            expect(res.body.message).to.be.equal("Invalid role");
+            return done();
+          });
+      });
+    });
```

Comment:
in this case we should throw 409 conflict

Reply:
Done please check in latest changes